evaluation/EDA/run_infer.py

```diff
@@ -118,13 +118,12 @@ def process_instance(
     instruction += AGENT_CLS_TO_INST_SUFFIX[agent.__class__.__name__]
 
     # Here's how you can run the agent (similar to the `main` function) and get the final task state
+    config.max_iterations = metadata.max_iterations
```

Comment:
This looks a bit strange to me. For most of the time, we really have treated the application configuration as immutable, or almost, even though it isn't. I think working that way had definite benefits, it makes it more understandable, and the user should be able to count on the values they set, once fully loaded and initialized, in the right order of precedence.

If we then change this at runtime, IMO it's harder for the users to understand what value came from where or why.

I wrote some more about this, but unfortunately I... lost the contents of that post. 😅 Anyway I guess that since we no longer let the core parse the command line args for the client scripts, the scripts have to do that. I wonder if we can do it better though, at some point (doesn't have to be right now).

Reply:
Yeah, I think creating an immutable config definitely has its benefits! But I think it could be slightly tricky to do now bc we might need to reload config from env var, `config.toml` etc.

I'm working on an evaluation refactor (should be done in the next couple days) for the new runtime, where I'll go through every `config` in the evaluation and initialize the whole config all at once (is this more aligned with what you think?):

https://github.com/OpenDevin/OpenDevin/blob/c3d6320f381c93e032598ea54bbeddfac4e83ac2/evaluation/swe_bench/run_infer.py#L117-L144

Reply:
> And no other place in SWE-bench sneaking in some other value in it, right? 😅

Yep! "sneaking in some other value" has caused me a lot of trouble a while back 😿  definitely want to get rid of that behavior

Reply:
Hey arguably, this line is sneaky too. 🤣 We can live with it since we're killing it again!

Reply:
Kill it in 2-5 days 🦾